### ISSUE TITLE
MyPy yield a false error with function(*args, x)

### ISSUE THREAD

**Issue by R-Peleg** [python/mypy#16677]
**Bug Report**

MyPy wrongly produce an error for "Too many argument" when calling a function with star syntax followed by an additional argument.

**To Reproduce**

```python

def foo(a: int, b: int, c: int, d: int) -> None:
    return None

lst = [1, 2, 3]
foo(*lst, 4)
```

**Expected Behavior**
MyPy should not yield any error, and the types are fit.

**Actual Behavior**
An error occurs:
```
mypy_reproduce.py:6: error: Too many arguments for "foo"  [call-arg]
```

**Your Environment**
- Mypy version used:1.4.1
- Mypy command-line flags: No flags
- Mypy configuration options from `mypy.ini` (and other config files): No config
- Python version used: 3.7.3
